File: ats-ness-front-prod/src/app/demo/models/Langues.ts
import {Candidat} from "./condidat";

export class Langues {
    id: number | undefined;
    name: string | undefined;
    lev: string | undefined;
    commentaire: string | undefined;

    // Relation avec l'entité Candidat
    candidat: Candidat | undefined;
}

File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/Langues.java
package nesstechnologies.fr.applicationesnbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString
public class Langues implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private  String name;
    private  String lev;
    private  String commentaire;

    @OneToMany(mappedBy = "langues", cascade = CascadeType.ALL)
    private List<Rating> ratings;

    @JsonIgnore
    @ManyToOne
    private Candidat candidat;
}
